logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -527,5 +527,41 @@ class DummyOutput2 < DummyOutput; end
         expect(value).to be > 0
       end
     end
+
+    context "when reloading a config that raises exception on pipeline.run" do
+      let(:new_config) { "input { generator { count => 10000 } }" }
+      let(:new_config_generator_counter) { 500 }
+
+      class BrokenGenerator < LogStash::Inputs::Generator
+        def register
+          raise ArgumentError
+        end
+      end
+
+      before :each do
+
+        allow(LogStash::Plugin).to receive(:lookup).with("input", "generator").and_return(BrokenGenerator)
+
+        File.open(config_path, "w") do |f|
+          f.write(new_config)
+          f.fsync
+        end
+
+        subject.send(:"reload_pipeline!", "main")
+      end
+
+      it "does not increase the successful reload count" do
+        snapshot = subject.metric.collector.snapshot_metric
+        value = snapshot.metric_store.get_with_path("/stats/pipelines")[:stats][:pipelines][:main][:reloads][:successes].value
+        expect(value).to eq(0)
+      end
```

Comment:
Well instead of making the trigger in the before :each you do it in the proc?

Something like this might work?

```ruby
expect { update_config }.to change { snapshot.metric_store.get_with_path("/stats/pipelines")[:stats][:pipelines][:main][:reloads][:successes].value }.by(1)
```


So rspec will do the following.

1. change proc
2. expect proc
3. change proc
4. apply the comparator matcher..